### PR TITLE
Add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by sentayhu19**
**In this PR**

- Class `Person` Created with  the following:
  - Instance vars: `@id`, `@name`, and `@age`.
  - Constructor with `name`,  `age`, and `parent_permission` as parameter. `name` and `parent_permission` are optional and have default values of `"Unknown"` and `true`.
  - Getters for `@id`, `@name`, and `@age`.
  - Setters for `@name` and `@age`.
  - Private method `is_of_age?` that returns `true` if `@age` is greater or equal to 18 and `false` otherwise.
  - Public method `can_use_services?` that returns `true` if person is of age or if they have permission from parents.
- Class `Student` created with the following:
  - Inherits from `Person`.
  - Constructor extends parent's constructor by adding `@classroom` and a parameter for it.
  - Method `play_hooky` that returns `"¯\(ツ)/¯"`.
- Class `Teacher` created with the following:
  - Inherits from `Person`.
  - Constructor extends parent's constructor by adding `@specialization` and a parameter for it.
  - Override `can_use_services?` so it always returns `true`.
- Each class should be saved in a separate file.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    @age
+  end
+  # setter methods
+
+  def name_setter(name)
+    @name = name
+  end
+
+  def age_setter(age)
+    @age = age
```
Comment (FrederickMih) [inline person.rb]: - [x] I think you don't longer require these methods after adding this line: `attr_accessor :name, :age` excluding Getter for `:id` which can equally be replace by` attr_reader :id`

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  end
+
+  def of_age?
+    @age >= 18
+  end
+  private :of_age?
+  def can_use_services?
+    of_age? || @parent_permission
+  end
```
Comment (FrederickMih) [inline person.rb:17]: - [x] `can_use_services?` should be a public method while `of_age?` should be a private method.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @sentayhu19 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

# To Highlight

-  PR has a title and summary :heavy_check_mark: 
-  GitHub workflow has been followed :heavy_check_mark: 
-  Linters checks are passing :heavy_check_mark: 


## Required Changes ♻️

- [x] ` README.md.` This is a solo project hence remove the other contributor in the Authors section of your README file.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lordkaito — APPROVED** [review]
Hi @sentayhu19, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**